src/app/core/services/asset-entry-builder.ts

```diff
@@ -1,28 +1,67 @@
-import {Injectable} from '@angular/core';
-import {AssetCreateRequest} from '@sovity.de/edc-client';
+import {Inject, Injectable} from '@angular/core';
+import {UiAssetCreateRequest} from '@sovity.de/edc-client';
 import {AssetEditorDialogFormValue} from '../../routes/connector-ui/asset-page/asset-create-dialog/asset-editor-dialog-form-model';
-import {AssetPropertyMapper} from './asset-property-mapper';
+import {APP_CONFIG, AppConfig} from '../config/app-config';
 import {DataAddressMapper} from './data-address-mapper';
 
 @Injectable()
 export class AssetEntryBuilder {
   constructor(
-    private assetPropertyMapper: AssetPropertyMapper,
+    @Inject(APP_CONFIG) private config: AppConfig,
     private dataAddressMapper: DataAddressMapper,
   ) {}
 
   /**
-   * Build {@link AssetCreateRequest} from {@link AssetEditorDialogFormValue}
+   * Build {@link UiAssetCreateRequest} from {@link AssetEditorDialogFormValue}
    *
    * @param formValue form value
    * @return asset create dto
    */
   buildAssetCreateRequest(
     formValue: AssetEditorDialogFormValue,
-  ): AssetCreateRequest {
-    let properties = this.assetPropertyMapper.buildProperties(formValue);
+  ): UiAssetCreateRequest {
+    let id = formValue.metadata?.id ?? 'no-id-was-set';
```

Comment:
I know it's copied. We should fix it to the right thing though, meaning adding a non-null assertion "!", as the null case should be dead code thanks to the form validation